utils/dataset_manifest/core.py

```diff
@@ -182,37 +193,69 @@ def step(self):
     def step(self, value):
         self._step = int(value)
 
-    def __iter__(self):
-        sources = (i for i in self._sources)
-        for idx in range(self._stop):
-            if idx in self.range_:
-                image = next(sources)
-                img = Image.open(image, mode='r')
+    def _get_img_properties(self, image: Union[str, BytesIO]) -> Dict[str, Any]:
+        img = Image.open(image, mode='r')
+        if self._data_dir:
+            img_name = os.path.relpath(image, self._data_dir)
+        else:
+            img_name = os.path.basename(image) if isinstance(image, str) else image.filename
 
-                img_name = os.path.relpath(image, self._data_dir) if self._data_dir \
-                    else os.path.basename(image) if not self._raw_data_used else image.filename
-                name, extension = os.path.splitext(img_name)
-                image_properties = {
-                    'name': name.replace('\\', '/'),
-                    'extension': extension,
-                }
+        name, extension = os.path.splitext(img_name)
+        image_properties = {
+            'name': name.replace('\\', '/'),
+            'extension': extension,
+        }
 
-                width, height = img.width, img.height
-                orientation = img.getexif().get(274, 1)
-                if orientation > 4:
-                    width, height = height, width
-                image_properties['width'] = width
-                image_properties['height'] = height
+        width, height = img.width, img.height
+        orientation = img.getexif().get(274, 1)
+        if orientation > 4:
+            width, height = height, width
+        image_properties['width'] = width
+        image_properties['height'] = height
 
-                if self._meta and img_name in self._meta:
-                    image_properties['meta'] = self._meta[img_name]
+        if self._meta and img_name in self._meta:
+            image_properties['meta'] = self._meta[img_name]
 
-                if self._use_image_hash:
-                    image_properties['checksum'] = md5_hash(img)
+        if self._use_image_hash:
+            image_properties['checksum'] = md5_hash(img)
 
-                yield image_properties
+        return image_properties
+
+    def __iter__(self):
+        if not self.are_sources_filtered:
```

Comment:
Consider replacing all these branches with just 1 algorithm instead. It will help to avoid problems such as updating only one branch when all of them need to be updated.